allennlp/common/testing/model_test_case.py

```diff
@@ -46,6 +46,14 @@ def ensure_model_can_train_save_and_load(self, param_file: str, tolerance: float
         reader = DatasetReader.from_params(params['dataset_reader'])
         iterator = DataIterator.from_params(params['iterator'])
 
+        model_dataset = reader.read(params['validation_data_path'])
```

Comment:
Do you need to duplicate these three lines?  Can you just move the gradient check to later in the method, after these objects are already created?